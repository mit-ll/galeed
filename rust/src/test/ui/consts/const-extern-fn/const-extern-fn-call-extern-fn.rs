#![feature(const_extern_fn)]

extern "C" {
    fn regular_in_block();
}

const extern fn bar() {
    unsafe {
        regular_in_block();
        //~^ ERROR: can only call other `const fn` within a `const fn`
    }
}

extern fn regular() {}

const extern fn foo() {
    unsafe {
        regular();
        //~^ ERROR: can only call other `const fn` within a `const fn`
    }
}

fn main() {}
