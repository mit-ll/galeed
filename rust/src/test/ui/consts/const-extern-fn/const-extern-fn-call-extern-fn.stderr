error[E0723]: can only call other `const fn` within a `const fn`, but `regular_in_block` is not stable as `const fn`
  --> $DIR/const-extern-fn-call-extern-fn.rs:9:9
   |
LL |         regular_in_block();
   |         ^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #57563 <https://github.com/rust-lang/rust/issues/57563> for more information
   = help: add `#![feature(const_fn)]` to the crate attributes to enable

error[E0723]: can only call other `const fn` within a `const fn`, but `regular` is not stable as `const fn`
  --> $DIR/const-extern-fn-call-extern-fn.rs:18:9
   |
LL |         regular();
   |         ^^^^^^^^^
   |
   = note: see issue #57563 <https://github.com/rust-lang/rust/issues/57563> for more information
   = help: add `#![feature(const_fn)]` to the crate attributes to enable

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0723`.
